src/main/java/monitoringsystemturbo/presenter/ExportPresenter.java

```diff
@@ -1,107 +1,178 @@
-                    if (allTimeCheckBox.isSelected()) {
-                        mainExporter.export(trackingService, applicationsToExport);
-                    } else {
-                        mainExporter.export(trackingService, applicationsToExport, fromTime, toTime);
-                    }
-                    Alert successAlert = new Alert(Alert.AlertType.INFORMATION);
-                    successAlert.setTitle("Success!");
-                    successAlert.setHeaderText(null);
-                    successAlert.setContentText("Data exported successfully! ");
-                    successAlert.showAndWait();
-                    primaryStage.close();
-                }
-            } catch (IOException e) {
-                Alert errorAlert = new Alert(Alert.AlertType.ERROR);
-                errorAlert.setTitle("Error!");
-                errorAlert.setHeaderText(null);
-                errorAlert.setContentText("Error occurred while exporting data.");
-                errorAlert.showAndWait();
-            }
-        }
-    }
-
-    @FXML
-    public void onCancel() {
-        primaryStage.close();
-    }
-
-    public void setPrimaryStage(Stage primaryStage) {
-        this.primaryStage = primaryStage;
-    }
-
-    public void setMainExporter(MainExporter mainExporter) {
-        this.mainExporter = mainExporter;
-    }
-
-    public void setTrackingService(TrackingService trackingService) {
-        this.trackingService = trackingService;
-    }
-}
+package monitoringsystemturbo.presenter;
+
+import com.jfoenix.controls.JFXDatePicker;
+import com.jfoenix.controls.JFXTimePicker;
+import javafx.fxml.FXML;
+import javafx.scene.control.Alert;
+import javafx.scene.control.CheckBox;
+import javafx.stage.Stage;
+import monitoringsystemturbo.controller.ConfirmExportController;
+import monitoringsystemturbo.exporter.MainExporter;
+import monitoringsystemturbo.model.TrackingService;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+
+import static java.time.LocalDate.now;
+
+public class ExportPresenter {
+
+    private Stage primaryStage;
+    private MainExporter mainExporter;
+    private TrackingService trackingService;
+
+    @FXML
+    private JFXDatePicker fromDatePicker;
+
+    @FXML
+    private JFXDatePicker toDatePicker;
+
+    @FXML
+    private JFXTimePicker toTimePicker;
+
+    @FXML
+    private JFXTimePicker fromTimePicker;
+
+    @FXML
+    private CheckBox wholeRangeCheckBox;
+
+    @FXML
+    private CheckBox fromBeginCheckBox;
+
+    @FXML
+    private CheckBox toNowCheckBox;
+
+    @FXML
+    public void initialize() {
+        fromDatePicker.setValue(now());
+        toDatePicker.setValue(now());
+        fromTimePicker.setValue(LocalTime.of(0, 0));
+        toTimePicker.setValue(LocalTime.now());
+    }
+
+    @FXML
+    public void onConfirm() {
```

Comment:
this is a really long method, consider splitting it into reasonable chunks(other methods) with good namings